src/main/java/de/tum/in/www1/artemis/web/rest/ResultResource.java

```diff
@@ -276,4 +279,48 @@ public ResponseEntity<Result> createResultForExternalSubmission(@PathVariable Lo
         return ResponseEntity.created(new URI("/api/results/" + savedResult.getId()))
                 .headers(HeaderUtil.createEntityCreationAlert(applicationName, true, ENTITY_NAME, savedResult.getId().toString())).body(savedResult);
     }
+
+    @GetMapping("exercises/{exerciseId}/feedback-details")
+    @EnforceAtLeastTutor
+    public ResponseEntity<FeedbackDetailsResponse> getAllFeedbackDetailsForExercise(@PathVariable Long exerciseId) {
+        log.debug("REST request to get all Feedback details for Exercise {}", exerciseId);
+        Exercise exercise = exerciseRepository.findByIdElseThrow(exerciseId);
+
+        // Get all Student participations for the exercise
+        Set<StudentParticipation> participations;
+        if (exercise.isTeamMode()) {
+            participations = studentParticipationRepository.findByExerciseIdWithLatestAndManualResultsWithTeamInformation(exercise.getId());
+        }
+        else {
+            participations = studentParticipationRepository.findByExerciseIdWithLatestAndManualResultsAndAssessmentNote(exercise.getId());
+        }
+        removeSubmissionAndExerciseData(participations);
```

Comment:
I do not think that this is the best approach. If I am not mistaken we are querying too much information from the database, then removing the information that we do not need. And then making another query to get information that could also have been included in the first query.

Why don't we simply add another query that directly includes the feedback? As I am not an expert on this topic you might want to get the opinion of somebody who is more familiar with this component / your advisor - maybe I am mistaken with the approach of writing 1 SQL query that directly returns the result that we need here.

Drawbacks of the SQL query that I see:
- if the way latest results are found is updated an additional query needs to be updated
- more complex query
Advantages
+ should have a better performance
+ the code in the resource would be quite simple